src/util/sharing_map.h

```diff
@@ -894,6 +894,8 @@ SHARING_MAPT2(, innert *)::get_container_node(const key_type &k)
     key >>= chunk;
   }
 
+  SM_ASSERT(ip->is_container());
```

Comment:
Why is it not `is_defined_container` here?

Reply:
The non-const version of `get_container_node()` adds empty nodes if the key doesn't exist yet in the map. In this case, `!empty()` will not hold.